src/main/java/sigma/Parser.java

```diff
@@ -0,0 +1,73 @@
+package sigma;
+
+import sigma.command.ByeCommand;
+import sigma.command.Commands;
+import sigma.command.DeadlineCommand;
+import sigma.command.DeleteCommand;
+import sigma.command.EventCommand;
+import sigma.command.ListCommand;
+import sigma.command.MarkCommand;
+import sigma.command.TodoCommand;
+import sigma.command.UnmarkCommand;
+import sigma.command.UnrecognisedCommand;
+import sigma.exception.SigmaException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a parser that parses user input.
+ */
+public class Parser {
+
+    /**
+     * Parses the user input and returns the corresponding command.
+     * @param command User input.
+     * @return Corresponding command.
+     */
+    public static Commands parse(String command) {
+        String[] split = command.split(" ", 2);
+        String cmd = split[0]; // Convert input to uppercase to match enum
+        switch (cmd) {
+        case "todo":
+            return new TodoCommand(split);
+        case "deadline":
+            return new DeadlineCommand(split);
+        case "event":
+            return new EventCommand(split);
+        case "list":
+            return new ListCommand(split);
+        case "mark":
+            return new MarkCommand(split);
+        case "unmark":
+            return new UnmarkCommand(split);
+        case "delete":
+            return new DeleteCommand(split);
+        case "bye":
+            return new ByeCommand(split);
+        case "find":
+            return new FindCommand(split);
+        default:
+            return new UnrecognisedCommand(split);
+        }
+    }
+
+    /**
+     * Parses the date and time input and returns a LocalDateTime object.
+     * @param timing Date and time input.
+     * @return LocalDateTime object.
+     * @throws SigmaException If the date and time input is invalid.
+     */
+    public static LocalDateTime dateTimeParse(String timing) throws SigmaException {
```

Comment:
`parseLocalDateTime` might be a better name for this method for convention and clarity.

Reply:
Good catch. Changed!